library/ccm.c

```diff
@@ -307,15 +388,33 @@ static int ccm_auth_crypt( mbedtls_ccm_context *ctx, int mode, size_t length,
                 break;
     }
 
+    if( len_left > 0 )
+    {
+        return( MBEDTLS_ERR_PLATFORM_FAULT_DETECTED );
+    }
     /*
      * Authentication: reset counter and crypt/mask internal tag
      */
+    flow_ctrl_local = 0;
     for( i = 0; i < q; i++ )
+    {
+        flow_ctrl_local++;
         ctr[15-i] = 0;
+    }
+    CHECK_FLOW_CTRL( q, fc6 );
 
     CTR_CRYPT( y, y, 16 );
+    CHECK_FLOW_CTRL( 16, fc7 );
     mbedtls_platform_memcpy( tag, y, tag_len );
 
+    if( length_dup != length || iv_dup != iv || iv_len_dup != iv_len ||
+        add_dup != add || add_len_dup != add_len || input_dup != input ||
+        output_dup != output || tag_dup != tag || tag_len_dup != tag_len ||
+        ( fc0 | fc1 | fc2 | fc3 | fc4 | fc5 | fc6 | fc7 ) != 0 )
+    {
+        return MBEDTLS_ERR_PLATFORM_FAULT_DETECTED;
```

Comment:
@AndrzejKurek , may be better to check in this loop that all values are fine and return 0?  And as default flow return error

Reply:
Sure, I'll change that

Reply:
Fixed :)